test/one/profiler/test/Runner.java

```diff
@@ -0,0 +1,242 @@
+            }
+            return possibleVersion;
+        }
+    }
+
+    private static Jvm detectJvm() throws RuntimeException{
+        // Example javaHome: /usr/lib/jvm/amazon-corretto-17.0.8.7.1-linux-x64
+        File javaHome = new File(System.getProperty("java.home"));
+
+        // J9 Check which relies on JDK 8 JRE file dir differences
+        File libDirectory = new File(javaHome, "lib");
+        Pattern j9IndicatorFilePattern = Pattern.compile("J9TraceFormat.dat");
+        File[] files = libDirectory.listFiles();
+        if (files != null) {
+            for (File file : files) {
+                if (j9IndicatorFilePattern.matcher(file.getName()).matches()) {
+                    return Jvm.J9;
+                }
+            }
+        }
+
+        // Zing check
+
+        // Workaround for java.home including /jre in JDK 8
+        if (currentJvmVersion == 8) {
+            javaHome = javaHome.getParentFile();
+        }
+
+        // Workaround for contents/home on mac
+        if (currentOs == Os.MACOS) {
+            javaHome = javaHome.getParentFile();
+        }
+        File etcDirectory = new File(javaHome, "etc");
+        File zing = new File(etcDirectory, "zing");
+        if (zing.exists()) {
+            return Jvm.ZING;
+        }
+
+        // Otherwise it's some variation of hotspot
+        return Jvm.HOTSPOT;
+    }
+
+    private static Arch detectArch() throws RuntimeException{
+        String architecture = System.getProperty("os.arch");
+
+        if (architecture.contains("x86_64") || architecture.contains("amd64")) {
+            return Arch.X64;
+        } else if (architecture.contains("aarch64")) {
+            return Arch.ARM64;
+        } else if (architecture.contains("arm")) {
+            return Arch.ARM32;
+        } else if (architecture.contains("ppc64le")) {
+            return Arch.PPC64LE;
+        } else if (architecture.endsWith("86")) {
+            return Arch.X86;
+        } else {
+            throw new RuntimeException("Unable to detect the OS used by the host.\nOS may be unsupported.");
+        }
+    }
+
+    public static boolean enabled(Test test) {
+        Os[] os = test.os();
+        int[] jdkVersion = test.jdkVersion();
+        Jvm[] jvm = test.jvm();
+        Arch[] arch = test.arch();
+
+        if (!test.enabled()) {
+            return false;
+        }
+
+        if (os.length > 0 && !Arrays.asList(os).contains(currentOs)) {
+            return false;
+        }
+
+        if (jdkVersion.length > 0 && !Arrays.asList(jdkVersion).contains(currentJvmVersion)) {
+            return false;
+        }
+
+        if (jvm.length > 0 && !Arrays.asList(jvm).contains(currentJvm)) {
+            return false;
+        }
+
+        if (arch.length > 0 && !Arrays.asList(arch).contains(currentArch)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public static boolean run(Method m, Set skipSet, boolean retainlogs) throws Exception{
```

Comment:
`Set<String>`
Also, instead of passing `skipSet` and `retainlogs` throughout multiple methods, I'd made them instance fields of `Runner`.

Reply:
Addressed. Will reflect in next commit